app/src/main/java/com/alphawallet/app/ui/WalletConnectActivity.java

```diff
@@ -791,14 +790,27 @@ private void onSessionRequest(Long id, WCPeerMeta peer, long chainId)
         chainIcon.bindData(chainIdOverride);
         remotePeerMeta = peer;
 
-        confirmationDialog = new ActionSheetDialog(this, peer, chainId, displayIcon, this);
-        confirmationDialog.show();
-        confirmationDialog.fullExpand();
+        confirmationDialog = new ActionSheetDialog(this, peer, chainIdOverride, displayIcon, this);
 
-        viewModel.track(Analytics.Action.WALLET_CONNECT_SESSION_REQUEST);
+        if (confirmationDialog.getActionSheetStatus() == ActionSheetStatus.ERROR_INVALID_CHAIN)
+        {
+            showErrorDialogUnsupportedNetwork(id, chainIdOverride);
+        }
+        else
+        {
+            confirmationDialog.show();
+            confirmationDialog.fullExpand();
+
+            viewModel.track(Analytics.Action.WALLET_CONNECT_SESSION_REQUEST);
+
+            if (!viewModel.isActiveNetwork(chainId) && !viewModel.isActiveNetwork(chainIdOverride))
+            {
+                openChainSelection();
+            }
+        }
     }
```

Comment:
Can this method be refactored to remove the nested conditionals? Maybe split it up a bit.